onset_maproom/tests/test_calc.py

```diff
@@ -5,6 +5,129 @@
 import data_test_calc
 
 
+def test_calibrate_available_water():
+
+    taw = xr.DataArray(data=[60, 40], dims="X", coords={"X": [1, 2]})
+    rho = xr.DataArray(
+        data=[0.5, 0.45], dims="Crop", coords={"Crop": ["Maize", "Rice"]}
+    )
+    raw = calc.calibrate_available_water(taw, rho)
+
+    assert (raw == [[30.0, 20.0], [27.0, 18.0]]).all()
+
+
+def test_single_stress_coeff():
+
+    et_crop = precip_sample() * 0 + 5
+    soil_moisture = xr.where(precip_sample() > 10, 31, 15)
+    taw = 60
+    raw = 0.5 * taw
+    ks = calc.single_stress_coeff(soil_moisture, taw, raw)
+
+    assert (ks.isel(T=[8, 10, 54, 58]) == 1).all()
+    assert (ks.dropna("T").where(lambda x: x != 1, drop=True) == 0.5).all()
+
+
+def test_hargreaves_et_ref():
+
+    tmin = data_test_calc.tmin_data_sample()
+    tmax = data_test_calc.tmax_data_sample()
+    temp_avg = (tmin + tmax) / 2
+    # For the record as we've seen data where tmin > tmax
+    temp_amp = (tmax - tmin).clip(min=0)
+    doy = tmin["T"].dt.dayofyear
+    lat = tmin["Y"]
+    if lat.units == "degree_north":
+        lat = lat * np.pi / 180
+        lat.attrs = dict(units="radian")
+    ra = calc.solar_radiation(doy, lat)
+    et_ref = calc.hargreaves_et_ref(temp_avg, temp_amp, ra)
+    expected = [
+        [[5.28141477, 5.27847271], [4.67106064, 4.56270918]],
+        [[5.67150704, 5.76156393], [5.18580189, 5.24758635]],
+        [[6.76812044, 6.43869058], [6.0802728, 5.67826723]],
+        [[6.42336374, 6.08255966], [6.28099376, 5.94158488]],
+    ]
+
+    assert np.allclose(et_ref, expected)
+
+
+def test_solar_radiation():
+
+    precip = data_test_calc.lat_time_data_sample()
+    doy = precip["T"].dt.dayofyear
+    lat = precip["Y"]
+    if lat.units == "degree_north":
+        lat = lat * np.pi / 180
+        lat.attrs = dict(units="radian")
+    ra = calc.solar_radiation(doy, lat)
+
+    assert (
+        [
+            ra.isel(T=0, Y=-1),
+            ra.isel(T=0, Y=-1),
+            ra.isel(T=0, Y=-1),
+            ra.isel(T=0, Y=0),
+            ra.isel(T=0, Y=0),
+        ]
+        < [
+            ra.isel(T=-1, Y=-1),
+            ra.isel(T=0, Y=0),
+            ra.isel(T=-1, Y=0),
+            ra.isel(T=-1, Y=0),
+            ra.isel(T=-1, Y=0),
+        ]
+    ).all()
```

Comment:
I haven't attempted to understand what you're testing here, but I strongly suspect that this code doesn't do what you meant. The things you're applying the `<` operator to are python lists, not numpy or xarray arrays, and the `<` operator on lists works completely diffferently. For example,
```
>>> [1, 2] < [2, 1]
True
```
Happy to go into more detail if you'd like, but the bottom line is don't try to do array math with lists.